airbyte/_executors/python.py

```diff
@@ -5,24 +5,69 @@
 import subprocess
 import sys
 from contextlib import suppress
+from functools import lru_cache
 from pathlib import Path
 from shutil import rmtree
 from typing import TYPE_CHECKING, Literal
 
+import requests
 from overrides import overrides
+from packaging.specifiers import SpecifierSet
+from packaging.version import Version
 from rich import print  # noqa: A004  # Allow shadowing the built-in
 
 from airbyte import exceptions as exc
 from airbyte._executors.base import Executor
 from airbyte._util.meta import is_windows
 from airbyte._util.telemetry import EventState, log_install_state
 from airbyte._util.venv_util import get_bin_dir
+from airbyte.constants import AIRBYTE_OFFLINE_MODE
+from airbyte.logs import warn_once
+from airbyte.version import get_version
 
 
 if TYPE_CHECKING:
     from airbyte.sources.registry import ConnectorMetadata
 
 
+@lru_cache(maxsize=128)
+def _get_pypi_python_requirements_cached(package_name: str) -> str | None:
+    """Get the requires_python field from PyPI for a package.
+
+    Args:
+        package_name: The PyPI package name to check
+
+    Returns:
+        The requires_python string from PyPI, or None if unavailable
+
+    Example:
+        For airbyte-source-hubspot, returns "<3.12,>=3.10"
+    """
+    if AIRBYTE_OFFLINE_MODE:
+        return None
+
+    url = f"https://pypi.org/pypi/{package_name}/json"
+    version = get_version()
+    response = requests.get(
+        url=url,
+        headers={"User-Agent": f"PyAirbyte/{version}" if version else "PyAirbyte"},
+        timeout=10,
+    )
+
+    if not response.ok:
+        return None
+
+    try:
+        data = response.json()
+        return data.get("info", {}).get("requires_python")
+    except Exception:
+        # Intentionally broad exception handling to ensure silent failure in all scenarios:
+        # - JSON parsing errors (json.JSONDecodeError)
+        # - Any other unexpected errors
+        # This ensures connector installation never fails due to version checking issues.
+        return None
```

Comment:
We've narrowed this enough, I think, that we should just use the supress() context manager with the exception(s) we want to ignore.

data: dict | None = None
with supress(SomeException):
    data = response.json()
if not data:
    return None
return data.get("info", {}).get("requires_python")